### DISCUSSION TITLE
Exporting filtered data from tables

### DISCUSSION THREAD

**Discussion by IvanDrag0** [Budibase/budibase#11397] [Help] [answered]
I have a table in my app that shows some filtered data that I would like to allow the user to export to CSV or JSON. I've looked at the [Export Data section](https://docs.budibase.com/docs/data-actions#export-data) of the documentation, however, my questions are (a) is it possible to export selected columns from the datasource instead of all of them/one of them and (b) is it possible to export all the rows shown in the table instead of having the user select what rows to export?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @IvanDrag0,

Currently the export action only works with the table component, additionally it only works with selected rows. To answer your question no I don't believe its possible to do what you require within Budibase. Feel free to create an enhancement to get this included as you describe. You can do this by using the link here; https://github.com/Budibase/budibase/issues/new/choose. If you need something done right away I'd recommend making a custom plugin which you could customise to suit your needs. https://docs.budibase.com/docs/custom-plugin

Additionally looking into creating a custom automation action may be what you are after. As you could pass whatever data the users sees back to automation and then use a custom automation action to handle the exporting to CSV.

I hope this helps.

**Reply by IvanDrag0** [reply]
@ConorWebb96 Thank you for the information!
